modules/integration/tests-integration/lazy-loading-test/src/test/java/org/wso2/appserver/integration/lazy/loading/LazyLoadingBaseTest.java

```diff
@@ -0,0 +1,486 @@
+            //Get the web app list
+            try {
+                webAppList = webAppAdminClient.getWebApplist(appName);
+                faultyWebAppList = webAppAdminClient.getFaultyWebAppList(appName);
+            } catch (RemoteException remoteException) {
+                String customErrorMessage = "remoteException Exception when calling methods in WebAppAdminClient";
+                log.error(customErrorMessage, remoteException);
+                throw new LazyLoadingTestException(customErrorMessage, remoteException);
+            }
+            // Find given app in faulty app list. If found return the loop with isWebAppDeployed=false
+            for (String faultWebAppName : faultyWebAppList) {
+                if (faultWebAppName.equalsIgnoreCase(appName)) {
+                    isWebAppDeployed = false;
+                    log.info(appName + "- Jaggery Application is faulty");
+                    doLoop = false;
+                    break;
+                }
+            }
+            // Find the given app in web app list. If found return the loop with isWebAppDeployed=true
+            for (String webAppName : webAppList) {
+                if (webAppName.equalsIgnoreCase(appName)) {
+                    isWebAppDeployed = true;
+                    log.info(appName + " Jaggery Application deployed in " + time + " millis");
+                    doLoop = false;
+                    break;
+                }
+            }
+            try {
+                //Sleep 500 milliseconds until the next verification of Jaggery application deployment.
+                Thread.sleep(500);
+            } catch (InterruptedException interruptedException) {
+                String customErrorMessage = "InterruptedException occurs when sleeping 500 milliseconds and while" +
+                        " waiting for Jaggery Application to get deployed ";
+                log.warn(customErrorMessage, interruptedException);
+            }
+
+        }
+        return isWebAppDeployed;
+    }
+
+    /**
+     * Check the given Carbon application is deployed correctly. This method is wait for 90 seconds
+     * for deployment of Carbon application and each 500 milliseconds  of wait it will check the
+     * deployment status.
+     *
+     * @param appName - Name of the application.
+     * @return boolean - true if the application is listed.
+     * @throws LazyLoadingTestException - Exception throws when creating WebAppAdminClient.
+     */
+    protected boolean isCarbonAppListed(String appName) throws LazyLoadingTestException {
+        ApplicationAdminClient appAdminClient;
+        String[] appList;
+        boolean isAppListed = false;
+        long time;
+        long startTime;
+        log.info("waiting for " + appName + " Carbon application to list. Max wait time :" + DEPLOYMENT_DELAY_IN_MILLISECONDS + " milliseconds");
+        try {
+            appAdminClient = new ApplicationAdminClient(backendURL, sessionCookie);
```

Comment:
@prabathabey  Yes, We need to create an instance of ApplicationAdminClient each time we invoke isCarbonAppListed method because it uses the latest session key.  If we move this to init method and use the instance that is created at init method it my have an expired session due to server restart at  test case level or we need to call this method  with a different user  session (may be different tenant)